### PR TITLE
Fix testcontainers/Docker incompatibility

### PR DESCRIPTION
**Pull request by djsstarburst**
Docker for Mac issued a new release that is incompatible
with the org.testcontainers version 1.14.3 library.  This
commit updates to testcontainers version 1.15.0-rc2, released
today, and fixes the consequences of that upgrade.

Here is the backtrace you'll see before these changes when you run a product test:

```
2020-10-01T14:16:30.173-0700	INFO	main	org.testcontainers.utility.RegistryAuthLocator	Credential helper/store (docker-credential-desktop) does not have credentials for index.docker.io
2020-10-01T14:16:30.629-0700	WARN	testcontainers-ryuk	org.testcontainers.utility.ResourceReaper	Can not connect to Ryuk at localhost:32768
java.net.SocketException: Broken pipe (Write failed)
	at java.base/java.net.SocketOutputStream.socketWrite0(Native Method)
	at java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110)
	at java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:138)
	at org.testcontainers.utility.ResourceReaper$FilterRegistry.register(ResourceReaper.java:424)
	at org.testcontainers.utility.ResourceReaper.lambda$null$1(ResourceReaper.java:152)
	at org.rnorth.ducttape.ratelimits.RateLimiter.doWhenReady(RateLimiter.java:27)
	at org.testcontainers.utility.ResourceReaper.lambda$start$2(ResourceReaper.java:136)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### CODE REVIEW

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java [hunk 1]
```diff
@@ -17,11 +17,11 @@
 import com.github.dockerjava.api.model.MemoryStatsConfig;
 import com.github.dockerjava.api.model.StatisticNetworksConfig;
 import com.github.dockerjava.api.model.Statistics;
-import com.github.dockerjava.core.InvocationBuilder;
 import io.airlift.log.Logger;
 import io.airlift.units.DataSize;
 import net.jodah.failsafe.FailsafeExecutor;
 import org.testcontainers.DockerClientFactory;
+import org.testcontainers.shaded.com.github.dockerjava.core.InvocationBuilder;
```
Comment (martint) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:24]: Why did this change? We shouldn't have dependencies on shaded classes, as they are an implementation detail of the library that uses them.

File: presto-druid/pom.xml [hunk 2]
```diff
@@ -155,6 +155,12 @@
             </exclusions>
         </dependency>
 
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (martint) [inline presto-druid/pom.xml:162]: Is this related to the upgrade?

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java (hunk 1 above)
Comment (djsstarburst) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java:24]: Agreed, but the package com.github.dockerjava.core no longer exists.  All I want is to be able to run product-tests again.  Suggestions?

File: presto-druid/pom.xml (hunk 2 above)
Comment (djsstarburst) [reply presto-druid/pom.xml:162]: Yes - - the org.testcontainers 1.14.3 jar depended on org.jetbrains.annotations, but the 1.15.0-rc2 jar does not.

**Comment by djsstarburst** [comment]
Actually, even with these changes, testcontainers doesn't start up correctly, though the docker containers do start up.

I'm closing this PR since it doesn't actually fix the problem of allowing folks to run product tests on Macs.

**Comment by martint** [comment]
> I'm closing this PR since it doesn't actually fix the problem of allowing folks to run product tests on Macs.

That's annoying, as the issue affects not only product tests, but a bunch of integration tests for various connectors (Elasticsearch, PostgreSQL, MySQL, Oracle, Cassandra, Druid, MongoDB, etc)

**Comment by djsstarburst** [comment]
> That's annoying, as the issue affects not only product tests, but a bunch of integration tests for various connectors (Elasticsearch, PostgreSQL, MySQL, Oracle, Cassandra, Druid, MongoDB, etc)

It's certainly annoying.  Tests run again after downgrading from Docker Desktop 2.4.0.0 to 2.3.0.5, found on this page: https://docs.docker.com/docker-for-mac/release-notes/

**Comment by wendigo** [comment]
@djsstarburst I was hit by that recently as well and downgraded Docker. I'll upgrade testcontainers and adjust our product tests when it's released.

**Comment by findepi** [comment]
has this been reported to testcontainers already? cc @bsideup

**Comment by bsideup** [comment]
@findepi yes, see https://github.com/testcontainers/testcontainers-java/issues/3166
